AmplifyPlugins/API/Sources/AWSAPIPlugin/Support/Utils/GraphQLRequest+toListQuery.swift

```diff
@@ -17,7 +17,10 @@ extension GraphQLRequest {
                                                    limit: Int? = nil,
                                                    nextToken: String? = nil,
                                                    apiName: String? = nil) -> GraphQLRequest<ResponseType> {
-        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelSchema: modelSchema, operationType: .query)
+        let primaryKeysOnly = ModelRegistry.modelType(from: modelSchema.name)?.rootPath != nil ? true : false
+        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelSchema: modelSchema,
+                                                               operationType: .query,
+                                                               primaryKeysOnly: primaryKeysOnly)
```

Comment:
this makes sense, we can move the line `let primaryKeysOnly = ModelRegistry.modelType(from: modelSchema.name)?.rootPath != nil ? true : false` into the initializer of `ModelBasedGraphQLDocumentBuilder`. `ModelBasedGraphQLDocumentBuilder`'s initializer already has the parameter `primaryKeysOnly` with a default value of false. We can expose another initializer for testing that removes the default parameter from the public initializer, and the implementation of the public initializer calls into the internal one with the value calculated from `ModelRegistry.modelType(from: modelSchema.name)?.rootPath != nil ? true : false`

Reply:
As i work on this, I've realized that we need to control it at the GraphQLRequest helpers of the sync based requests `GraphQLRequest+AnyModelWithSync` used by DataStore. DataStore should be backwards compatible with older versions of the app, ie. should always set `primaryKeysOnly` to false so that the eager loaded selection set is included, thus new apps send mutations that work with old apps on older versions of DataStore.

Reply:
Actually the same can be said for apps using API directly, the selection set change in the mutation request makes is so that the subscriptions running on the old app receives data that is no longer eager loaded, which will fail to decode to the old codegen types

Reply:
So, the initializer for `ModelBasedGraphQLDocumentBuilder` should have the `primaryKeysOnly` parameter, not as part of the internal init for testing, to control it to be `false` for DataStore requests if we want remain eager loading data into the LazyReference types.

Reply:
updated this